clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -404,6 +404,65 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
                                  mlir::ValueRange value = {}) {
     return create<mlir::cir::YieldOp>(loc, value);
   }
+
+  mlir::cir::CallOp
+  createCallOp(mlir::Location loc,
+               mlir::SymbolRefAttr callee = mlir::SymbolRefAttr(),
+               mlir::Type returnType = mlir::cir::VoidType(),
+               mlir::ValueRange operands = mlir::ValueRange(),
+               mlir::cir::ExtraFuncAttributesAttr extraFnAttr = {}) {
+
+    mlir::cir::CallOp callOp =
+        create<mlir::cir::CallOp>(loc, callee, returnType, operands);
+
+    if (extraFnAttr) {
+      callOp->setAttr("extra_attrs", extraFnAttr);
+    } else {
+      mlir::NamedAttrList empty;
+      callOp->setAttr("extra_attrs",
+                      mlir::cir::ExtraFuncAttributesAttr::get(
+                          getContext(), empty.getDictionary(getContext())));
+    }
+    return callOp;
+  }
+
+  mlir::cir::CallOp
+  createCallOp(mlir::Location loc, mlir::cir::FuncOp callee,
+               mlir::ValueRange operands = mlir::ValueRange(),
+               mlir::cir::ExtraFuncAttributesAttr extraFnAttr = {}) {
+    mlir::Type returnType = mlir::cir::VoidType();
+    if (!callee.getFunctionType().isVoid()) {
```

Comment:
I rather we don't have to do all the return type stuff logic here, it should be done automatically by a tablegen builder, like we were doing before.